test/common/formatter/substitution_formatter_test.cc

```diff
@@ -2401,8 +2402,83 @@ TEST(SubstitutionFormatterTest, GrpcStatusFormatterTest) {
   }
 }
 
-TEST(SubstitutionFormatterTest, GrpcStatusNumberFormatterTest) {
-  GrpcStatusFormatter formatter("grpc-status", "", absl::optional<size_t>(), true);
+TEST(SubstitutionFormatterTest, GrpcStatusFormatterSnakeStringTest) {
+  GrpcStatusFormatter formatter("grpc-status", "", absl::optional<size_t>(),
+                                GrpcStatusFormatter::Format::SnakeString);
+  NiceMock<StreamInfo::MockStreamInfo> stream_info;
+  Http::TestRequestHeaderMapImpl request_header;
+  Http::TestResponseHeaderMapImpl response_header;
+  Http::TestResponseTrailerMapImpl response_trailer;
+  std::string body;
+
+  std::array<std::string, 17> grpc_statuses{"OK",
```

Comment:
Use `std::vector`, so you do not have to specify the size. The rest of the code should stay the same.

Reply:
Done.